RFC/src/RFC-0340_VNConsensusOverview.md

```diff
@@ -1,6 +1,244 @@
+performant.
+
+There is a good survey of consensus mechanisms on
+[Tari Labs University](https://tlu.tarilabs.com/consensus-mechanisms/consensus-mechanisms.html).
+
+From the point of view of a DAN committee, the ideal consensus algorithm is one that
+1. allows a high number of transactions per second, and does does have unnecessary pauses (i.e. a partially synchronous
+   or asynchronous model).
+2. Is Byzantine Fault tolerant.
+3. Is relatively efficient from a network communication point of view (number of messages passed per state agreement).
+4. Is relatively simple to implement (to reduce the bug and vulnerability surface in implementations).
+
+A summary of some of the most well-known BFT algorithms is presented in
+[this table](https://tlu.tarilabs.com/consensus-mechanisms/BFT-consensus-mechanisms-applications/MainReport.html#summary-of-findings).
+
+A close reading of the algorithms presented suggest that [LinBFT](https://arxiv.org/pdf/1807.01829.pdf), which is based
+on [HotStuff] BFT provide the best trade-offs for the goals that a DAN committee is trying to achieve:
+1. The algorithm is optimistic, i.e. as soon as quorum is reached on a particular state, the committee can move onto the
+   next one. There is no need to wait for the "timeout" period as we do in e.g. Tendermint. This allows instructions to
+   be executed alomst as quickly as they are received.
+2. The algorithm is efficient in communication, requiring O(n) messages per state agreement in most practical cases.
+   This is compared to e.g. PBFT which requires O(n<sup>4</sup>) messages.
+3. The algorithm is modular and relatively simple to implement.
+
+Potential drawbacks to using HotStuff include:
+1. Each round required the election of a _leader_. Having a leader dramatically simplifies the consensus algorithm; it
+   allows a linear number of messages to be sent between the leader and the other replicas in order to agree on the
+   current state; and it allows a strict ordering to be established on instructions without having to resort to e.g.
+   proof of work. However, if the choice of leader is deterministic, attackers can identify and potentially DDOS the
+   leader for a given round, causing the algorithm to time out. There are ways to mitigate this attack for a _specific
+   round_, as suggested in the LinBFT paper, such as using Verifiable Random Functions, but DDOSing a single replica
+   means that, on average, the algorithm will time out every 1/n rounds.
+2. The attack described above only pauses progress in Hotstuff for the timeout period. In similar protocols, e.g.
+   Tendermint it can be shown to [delay progress indefinitely](https://arxiv.org/pdf/1803.05069).
+
+Given these trade-offs, there is strong evidence to suggest that [HotStuff] BFT, when implemented on the Tari DAN will
+provide BFT security guarantees with liveness performance in the sub-second scale and throughput on the order of
+thousands of instructions per second.
```

Comment:
`thousands of instructions per second` is perhaps too strong a claim. Any real-world references? Or is this a reference to the perceived collective performance of all DAN committees together?

Reply:
Yes, the HotStuff paper has benchmarks.

Reply:
Understood. I saw the performance graphs in the paper are showing kilo operations per second, but in a lab environment, with enough outstanding requests ("_instructions_") to keep their batches full.